### PR TITLE
Updates API ref for jobs. Adds doc on Kubernetes Scheduler persistent

### LINKED ISSUE #3915: Distributed Scheduler API building block docs
### Jobs pt 1 PR
_merged_ #4164 

- [x] Overview for Jobs API
- [x] Add to conceptual overview
- [x] Add to building blocks concepts
- [x] Update preview feature doc
- [x] Update list of Alpha APIs
- [x] Add API reference doc
- [x] Diagrams
- [x] Update `dapr init` doc to include Scheduler service as included

### Jobs pt 2 PR
_merged_ #4240 

- [x] Jobs API how-to - minor add from README in Go-SDK example PR
- [x] Check formatting and grammar (ie: Jobs overview)

### Jobs pt 3 PR
_in progress_ #4243 

- [x] Screenshot of `dapr init` and `docker ps` output that shows scheduler service is running
- [x] Finish how-to based on future quickstart or example
- [x] Add how to consume a triggered job

**The associated pull request from dapr/dapr, dapr/components-contrib, or other Dapr code repos**
dapr/proposals#44

### PR DESCRIPTION
**Pull request by JoshVanL**

### CODE REVIEW

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+**Job data will be lost during these events**.
+
+To make the Scheduler data resilient to restarts, a persistent volume must be mounted to the Scheduler StatefulSet.
+This persistent volume is backed by a real disk that is provided by the hosted Cloud Provider or Kubernetes infrastructure platform.
+The size this disk should be is determined by how many jobs are expected to be persisted at once however, 64Gb should be more than sufficient for most use cases.
+Some Kubernetes providers recommend using a [CSI driver](https://kubernetes.io/docs/concepts/storage/volumes/#csi) to provision the underlying disks.
+Below are a list of useful links to the relevant documentation for creating a persistent disk for the major cloud providers:
+- [Google Cloud Persistent Disk](https://cloud.google.com/compute/docs/disks)
+- [Amazon EBS Volumes](https://aws.amazon.com/blogs/storage/persistent-storage-for-kubernetes/)
+- [Azure AKS Storage Options](https://learn.microsoft.com/en-us/azure/aks/concepts-storage)
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md]:
```suggestion
- [Azure AKS Storage Options](https://learn.microsoft.com/azure/aks/concepts-storage)
```

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+- [Digital Ocean Block Storage](https://www.digitalocean.com/docs/kubernetes/how-to/add-volumes/)
+- [VMWare vSphere Storage](https://docs.vmware.com/en/VMware-vSphere/7.0/vmware-vsphere-with-tanzu/GUID-A19F6480-40DC-4343-A5A9-A5D3BFC0742E.html)
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md]:
```suggestion
- [VMWare vSphere Storage](https://docs.vmware.com/VMware-vSphere/7.0/vmware-vsphere-with-tanzu/GUID-A19F6480-40DC-4343-A5A9-A5D3BFC0742E.html)
```

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+- [OpenShift Persistent Storage](https://docs.openshift.com/container-platform/4.6/storage/persistent_storage/persistent-storage-aws-efs.html)
+- [Alibaba Cloud Disk Storage](https://www.alibabacloud.com/help/en/ack/ack-managed-and-ack-dedicated/user-guide/create-a-pvc)
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md]:
```suggestion
- [Alibaba Cloud Disk Storage](https://www.alibabacloud.com/help/ack/ack-managed-and-ack-dedicated/user-guide/create-a-pvc)
```

**Review by hhunter-ms — COMMENTED** [review]
just removing localized links to clear error

File: daprdocs/content/en/developing-applications/building-blocks/jobs/howto-schedule-jobs.md [hunk 4]
```diff
@@ -8,19 +8,22 @@ description: "Learn how to use the jobs API to schedule jobs"
...
+Include a diagram or image, if possible.
 -->
 
-
-
 ## Set up the Scheduler service
 
-When you run `dapr init` in either self-hosted mode or on Kubernetes, the Dapr scheduler service is started. 
+{{% alert title="Warning" color="warning" %}}
+By default, Job data is not resilient to [Scheduler]({{< ref scheduler.md >}}) service restarts.
```
Comment (hhunter-ms) [inline daprdocs/content/en/developing-applications/building-blocks/jobs/howto-schedule-jobs.md]:
```suggestion
By default, job data is not resilient to [Scheduler]({{< ref scheduler.md >}}) service restarts.
```

File: daprdocs/content/en/developing-applications/building-blocks/jobs/jobs-overview.md [hunk 5]
```diff
@@ -8,14 +8,19 @@ description: "Overview of the jobs API building block"
...
 
 Jobs in Dapr consist of:
-- The jobs API building block
+- [The jobs API building block]({{< ref jobs_api.md >}})
 - [The Scheduler control plane service]({{< ref "concepts/dapr-services/scheduler.md" >}})
 
 [See example scenarios.]({{< ref "#scenarios" >}})
 
+{{% alert title="Warning" color="warning" %}}
+By default, Job data is not resilient to [Scheduler]({{< ref scheduler.md >}}) service restarts.
```
Comment (hhunter-ms) [inline daprdocs/content/en/developing-applications/building-blocks/jobs/jobs-overview.md]:
```suggestion
By default, job data is not resilient to [Scheduler]({{< ref scheduler.md >}}) service restarts.
```

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
+---
+type: docs
+title: "How-to: Persist Scheduler Jobs"
+linkTitle: "How-to: Persist Scheduler Jobs"
+weight: 50000
+description: "Configure Scheduler to persist its database to make it resilient to restarts"
+---
+
+The [Scheduler]({{< ref scheduler.md >}}) service is responsible for writing Jobs to its embedded database and scheduling them for execution.
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md]:
```suggestion
The [Scheduler]({{< ref scheduler.md >}}) service is responsible for writing jobs to its embedded database and scheduling them for execution.
```

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+By default, the Scheduler service database writes this data to an in-memory ephemeral tempfs volume, meaning that this **data is not persisted across restarts**.
+**Job data will be lost during these events**.
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md]:
```suggestion
```

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
+---
+type: docs
+title: "How-to: Persist Scheduler Jobs"
+linkTitle: "How-to: Persist Scheduler Jobs"
+weight: 50000
+description: "Configure Scheduler to persist its database to make it resilient to restarts"
+---
+
+The [Scheduler]({{< ref scheduler.md >}}) service is responsible for writing Jobs to its embedded database and scheduling them for execution.
+By default, the Scheduler service database writes this data to an in-memory ephemeral tempfs volume, meaning that this **data is not persisted across restarts**.
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md]:
```suggestion
By default, the Scheduler service database writes this data to an in-memory ephemeral tempfs volume, meaning that **this data is not persisted across restarts**. Job data will be lost during these events.
```

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+**Job data will be lost during these events**.
+
+To make the Scheduler data resilient to restarts, a persistent volume must be mounted to the Scheduler StatefulSet.
+This persistent volume is backed by a real disk that is provided by the hosted Cloud Provider or Kubernetes infrastructure platform.
+The size this disk should be is determined by how many jobs are expected to be persisted at once however, 64Gb should be more than sufficient for most use cases.
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md]:
```suggestion
Disk size is determined by how many jobs are expected to be persisted at once; however, 64Gb should be more than sufficient for most use cases.
```

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+linkTitle: "How-to: Persist Scheduler Jobs"
+weight: 50000
+description: "Configure Scheduler to persist its database to make it resilient to restarts"
+---
+
+The [Scheduler]({{< ref scheduler.md >}}) service is responsible for writing Jobs to its embedded database and scheduling them for execution.
+By default, the Scheduler service database writes this data to an in-memory ephemeral tempfs volume, meaning that this **data is not persisted across restarts**.
+**Job data will be lost during these events**.
+
+To make the Scheduler data resilient to restarts, a persistent volume must be mounted to the Scheduler StatefulSet.
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md]:
```suggestion
To make the Scheduler data resilient to restarts, a persistent volume must be mounted to the Scheduler `StatefulSet`.
```

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+- [Google Cloud Persistent Disk](https://cloud.google.com/compute/docs/disks)
+- [Amazon EBS Volumes](https://aws.amazon.com/blogs/storage/persistent-storage-for-kubernetes/)
+- [Azure AKS Storage Options](https://learn.microsoft.com/azure/aks/concepts-storage)
+- [Digital Ocean Block Storage](https://www.digitalocean.com/docs/kubernetes/how-to/add-volumes/)
+- [VMWare vSphere Storage](https://docs.vmware.com/VMware-vSphere/7.0/vmware-vsphere-with-tanzu/GUID-A19F6480-40DC-4343-A5A9-A5D3BFC0742E.html)
+- [OpenShift Persistent Storage](https://docs.openshift.com/container-platform/4.6/storage/persistent_storage/persistent-storage-aws-efs.html)
+- [Alibaba Cloud Disk Storage](https://www.alibabacloud.com/help/ack/ack-managed-and-ack-dedicated/user-guide/create-a-pvc)
+
+
+Once the persistent volume class has been made available, Dapr can be installed with the following command with Scheduler configured to use the persistent volume class (replace `my-storage-class` with the name of the storage class):
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md]:
```suggestion
Once the persistent volume class is available, you can install Dapr using the following command, with Scheduler configured to use the persistent volume class (replace `my-storage-class` with the name of the storage class):
```

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+
+> If Dapr is already installed, the control plane will need to be completely [uninstalled]({{< ref dapr-uninstall.md >}}) in order for the Scheduler StatefulSet to be recreated with the new persistent volume.
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md]:
```suggestion
{{% alert title="Note" color="primary" %}}
If Dapr is already installed, the control plane needs to be completely [uninstalled]({{< ref dapr-uninstall.md >}}) in order for the Scheduler `StatefulSet` to be recreated with the new persistent volume.
{{% /alert %}}
```

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+
+{{< tabs "Dapr CLI" "Helm" >}}
+ <!-- Dapr CLI -->
+{{% codetab %}}
+
+```bash
+    dapr init -k --set dapr_scheduler.cluster.storageClassName=my-storage-class
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md]:
```suggestion
dapr init -k --set dapr_scheduler.cluster.storageClassName=my-storage-class
```

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+```
+
+{{% /codetab %}}
+
+ <!-- Helm -->
+{{% codetab %}}
+
+```bash
+    helm upgrade --install dapr dapr/dapr \
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md]:
```suggestion
helm upgrade --install dapr dapr/dapr \
```

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+    --version={{% dapr-latest-version short="true" %}} \
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md]:
```suggestion
--version={{% dapr-latest-version short="true" %}} \
```

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+    --namespace dapr-system \
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md]:
```suggestion
--namespace dapr-system \
```

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+    --create-namespace \
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md]:
```suggestion
--create-namespace \
```

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+    --set dapr_scheduler.cluster.storageClassName=my-storage-class \
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md]:
```suggestion
--set dapr_scheduler.cluster.storageClassName=my-storage-class \
```

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+    --wait
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-persisting-scheduler.md]:
```suggestion
--wait
```

File: daprdocs/content/en/operations/hosting/self-hosted/self-hosted-persisting-scheduler.md [hunk 20]
```diff
@@ -0,0 +1,18 @@
+---
+type: docs
+title: "How-to: Persist Scheduler Jobs"
+linkTitle: "How-to: Persist Scheduler Jobs"
+weight: 50000
+description: "Configure Scheduler to persist its database to make it resilient to restarts"
+---
+
+The [Scheduler]({{< ref scheduler.md >}}) service is responsible for writing Jobs to its embedded database and scheduling them for execution.
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/hosting/self-hosted/self-hosted-persisting-scheduler.md]:
```suggestion
The [Scheduler]({{< ref scheduler.md >}}) service is responsible for writing jobs to its embedded database and scheduling them for execution.
```

File: daprdocs/content/en/operations/hosting/self-hosted/self-hosted-persisting-scheduler.md [hunk 21]
```diff
@@ -0,0 +1,18 @@
...
+By default, the Scheduler service database writes this data to the local volume `dapr_scheduler`, meaning that this **data is persisted across restarts**.
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/hosting/self-hosted/self-hosted-persisting-scheduler.md]:
```suggestion
By default, the Scheduler service database writes this data to the local volume `dapr_scheduler`, meaning that **this data is persisted across restarts**.
```

File: daprdocs/content/en/operations/hosting/self-hosted/self-hosted-persisting-scheduler.md [hunk 22]
```diff
@@ -0,0 +1,18 @@
...
+
+The Scheduler persistent volume can be modified with a custom volume that is pre-existing, or will be created by Dapr.
+
+> If Dapr is already installed, the control plane will need to be completely [uninstalled]({{< ref dapr-uninstall.md >}}) in order for the Scheduler container to be recreated with the new persistent volume.
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/hosting/self-hosted/self-hosted-persisting-scheduler.md]:
```suggestion
{{% alert title="Note" color="primary" %}}
If Dapr is already installed, the control plane needs to be completely [uninstalled]({{< ref dapr-uninstall.md >}}) in order for the Scheduler container to be recreated with the new persistent volume.
{{% /alert %}}
```

File: daprdocs/content/en/operations/hosting/self-hosted/self-hosted-persisting-scheduler.md [hunk 23]
```diff
@@ -0,0 +1,18 @@
...
+
+```bash
+    dapr init --scheduler-volume my-scheduler-volume
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/hosting/self-hosted/self-hosted-persisting-scheduler.md]:
```suggestion
dapr init --scheduler-volume my-scheduler-volume
```

File: daprdocs/content/en/reference/api/jobs_api.md [hunk 24]
```diff
@@ -10,8 +10,16 @@ weight: 1300
 The jobs API is currently in alpha.
 {{% /alert %}}
 
+{{% alert title="Warning" color="warning" %}}
+By default, Job data is not resilient to [Scheduler]({{< ref scheduler.md >}}) service restarts.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/api/jobs_api.md]:
```suggestion
By default, job data is not resilient to [Scheduler]({{< ref scheduler.md >}}) service restarts.
```

File: daprdocs/content/en/reference/api/jobs_api.md [hunk 25]
```diff
@@ -22,22 +30,48 @@ POST http://localhost:3500/v1.0-alpha1/jobs/<name>
 
 ### URL parameters
 
+> **At least one of `schedule` or `dueTime` must be provided, but they can also be provided together**.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/api/jobs_api.md]:
```suggestion
{{% alert title="Note" color="primary" %}}
At least one of `schedule` or `dueTime` must be provided, but they can also be provided together.
{{% /alert %}}
```

File: daprdocs/content/en/reference/api/jobs_api.md [hunk 26]
```diff
@@ -22,22 +30,48 @@ POST http://localhost:3500/v1.0-alpha1/jobs/<name>
...
+
 Parameter | Description
 --------- | -----------
 `name` | Name of the job you're scheduling
-`data` | A string value and can be any related content. Content is returned when the reminder expires. For example, this may be useful for returning a URL or anything related to the content.
-`dueTime` | Specifies the time after which this job is invoked. Its format should be [time.ParseDuration](https://pkg.go.dev/time#ParseDuration)
+`data` | A protobuf message `@type` `value` pair. `@type` must be of a [well-known type](https://protobuf.dev/reference/protobuf/google.protobuf). Value is the serialized data.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/api/jobs_api.md]:
```suggestion
`data` | A protobuf message `@type`/`value` pair. `@type` must be of a [well-known type](https://protobuf.dev/reference/protobuf/google.protobuf). `value` is the serialized data.
```

File: daprdocs/content/en/reference/api/jobs_api.md [hunk 27]
```diff
@@ -22,22 +30,48 @@ POST http://localhost:3500/v1.0-alpha1/jobs/<name>
...
+`schedule` | schedule is an optional schedule at which the job is to be run. Details of the format are below.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/api/jobs_api.md]:
```suggestion
`schedule` | An optional schedule at which the job is to be run. Details of the format are below.
```

File: daprdocs/content/en/reference/api/jobs_api.md [hunk 28]
```diff
@@ -22,22 +30,48 @@ POST http://localhost:3500/v1.0-alpha1/jobs/<name>
...
+`dueTime` | dueTime is the optional time at which the job should be active, or the "one shot" time if other scheduling type fields are not provided. Accepts a "point in time" string in the format of RFC3339, Go duration string (calculated from creation time), or non-repeating ISO8601.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/api/jobs_api.md]:
```suggestion
`dueTime` | An optional time at which the job should be active, or the "one shot" time, if other scheduling type fields are not provided. Accepts a "point in time" string in the format of RFC3339, Go duration string (calculated from creation time), or non-repeating ISO8601.
```

File: daprdocs/content/en/reference/api/jobs_api.md [hunk 29]
```diff
@@ -22,22 +30,48 @@ POST http://localhost:3500/v1.0-alpha1/jobs/<name>
...
+`repeats` | repeats is the optional number of times in which the job should be triggered. If not set, the job will run indefinitely or until expiration.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/api/jobs_api.md]:
```suggestion
`repeats` | An optional number of times in which the job should be triggered. If not set, the job runs indefinitely or until expiration.
```

File: daprdocs/content/en/reference/api/jobs_api.md [hunk 30]
```diff
@@ -22,22 +30,48 @@ POST http://localhost:3500/v1.0-alpha1/jobs/<name>
...
+`ttl` | ttl is the optional time to live or expiration of the job. Accepts a "point in time" string in the format of RFC3339, Go duration string (calculated from job creation time), or non-repeating ISO8601.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/api/jobs_api.md]:
```suggestion
`ttl` | An optional time to live or expiration of the job. Accepts a "point in time" string in the format of RFC3339, Go duration string (calculated from job creation time), or non-repeating ISO8601.
```

File: daprdocs/content/en/reference/api/jobs_api.md [hunk 31]
```diff
@@ -22,22 +30,48 @@ POST http://localhost:3500/v1.0-alpha1/jobs/<name>
...
+
+#### schedule
+`schedule` accepts both systemd timer style cron expressions, as well as human readable '@' prefixed period strings as defined below.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/api/jobs_api.md]:
```suggestion
`schedule` accepts both systemd timer-style cron expressions, as well as human readable '@' prefixed period strings, as defined below.
```

File: daprdocs/content/en/reference/cli/dapr-init.md [hunk 32]
```diff
@@ -45,6 +45,7 @@ dapr init [flags]
 |        N/A            | DAPR_HELM_REPO_PASSWORD | A password for a private Helm chart  |The password required to access the private Dapr Helm chart. If it can be accessed publicly, this env variable does not need to be set| |
 |  `--container-runtime`  |              |    `docker`      | Used to pass in a different container runtime other than Docker. Supported container runtimes are: `docker`, `podman` |
 |  `--dev`  |              |          | Creates Redis and Zipkin deployments when run in Kubernetes. |
+|  `--scheduler-volume`  |              |          | Self-hosted only. Optionally specify a volume for the scheduler service data directory. By default, scheduler data is not persisted and not resilient to restarts without this flag. |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/cli/dapr-init.md]:
```suggestion
|  `--scheduler-volume`  |              |          | Self-hosted only. Optionally, you can specify a volume for the scheduler service data directory. By default, without this flag, scheduler data is not persisted and not resilient to restarts. |
```

**Review by hhunter-ms — CHANGES_REQUESTED** [review]
Initial review

File: daprdocs/content/en/reference/api/jobs_api.md [hunk 33]
```diff
@@ -46,24 +82,26 @@ Parameter | Description
...
+  -H "Content-Type: application/json"
   -d '{
         "job": {
             "data": {
-                "HanSolo": "Running spice"
+	            "@type": "type.googleapis.com/google.protobuf.StringValue",
+	            "value": "Running spice"
             },
-            "dueTime": "30s"
+            "scheudle": "@every 1m",
```
Comment (cicoyle) [inline daprdocs/content/en/reference/api/jobs_api.md]:
```suggestion
            "schedule": "@every 1m",
```

**Review by cicoyle — APPROVED** [review]
one spelling nit, but otherwise looks great 🎉

File: daprdocs/content/en/operations/hosting/self-hosted/self-hosted-persisting-scheduler.md [hunk 34]
```diff
@@ -0,0 +1,22 @@
...
+weight: 50000
+description: "Configure Scheduler to persist its database to make it resilient to restarts"
+---
+
+The [Scheduler]({{< ref scheduler.md >}}) service is responsible for writing jobs to its embedded database and scheduling them for execution.
+By default, the Scheduler service database writes this data to the local volume `dapr_scheduler`, meaning that **this data is persisted across restarts**.
+
+The host file location for this local volume will typically be located at either `/var/lib/docker/volumes/dapr_scheduler/_data` or `~/.local/share/containers/storage/volumes/dapr_scheduler/_data`, depending on your container runtime.
+
+The Scheduler persistent volume can be modified with a custom volume that is pre-existing, or will be created by Dapr.
```
Comment (msfussell) [inline daprdocs/content/en/operations/hosting/self-hosted/self-hosted-persisting-scheduler.md]:
```suggestion
The Scheduler persistent volume can be modified with a custom volume that is pre-existing, or is created by Dapr.
```

File: daprdocs/content/en/operations/hosting/self-hosted/self-hosted-persisting-scheduler.md [hunk 35]
```diff
@@ -0,0 +1,22 @@
...
+The host file location for this local volume will typically be located at either `/var/lib/docker/volumes/dapr_scheduler/_data` or `~/.local/share/containers/storage/volumes/dapr_scheduler/_data`, depending on your container runtime.
+
+The Scheduler persistent volume can be modified with a custom volume that is pre-existing, or is created by Dapr.
+
+{{% alert title="Note" color="primary" %}}
+If Dapr is already installed, the control plane needs to be completely [uninstalled]({{< ref dapr-uninstall.md >}}) in order for the Scheduler container to be recreated with the new persistent volume.
+{{% /alert %}}
+
+```bash
+dapr init --scheduler-volume my-scheduler-volume
```
Comment (msfussell) [inline daprdocs/content/en/operations/hosting/self-hosted/self-hosted-persisting-scheduler.md:26]: What is not clear is whether `--scheduler-volume my-scheduler-volume` is the default if all I do is `dapr init`. Currently it looks like it is not.

File: daprdocs/content/en/operations/hosting/self-hosted/self-hosted-persisting-scheduler.md [hunk 36]
```diff
@@ -0,0 +1,22 @@
...
+
+The [Scheduler]({{< ref scheduler.md >}}) service is responsible for writing jobs to its embedded database and scheduling them for execution.
+By default, the Scheduler service database writes this data to the local volume `dapr_scheduler`, meaning that **this data is persisted across restarts**.
+
+The host file location for this local volume will typically be located at either `/var/lib/docker/volumes/dapr_scheduler/_data` or `~/.local/share/containers/storage/volumes/dapr_scheduler/_data`, depending on your container runtime.
+
+The Scheduler persistent volume can be modified with a custom volume that is pre-existing, or is created by Dapr.
+
+{{% alert title="Note" color="primary" %}}
+If Dapr is already installed, the control plane needs to be completely [uninstalled]({{< ref dapr-uninstall.md >}}) in order for the Scheduler container to be recreated with the new persistent volume.
```
Comment (msfussell) [inline daprdocs/content/en/operations/hosting/self-hosted/self-hosted-persisting-scheduler.md]:
```suggestion
By default `dapr init` creates a local persistent volume on your drive. If Dapr is already installed, the control plane needs to be completely [uninstalled]({{< ref dapr-uninstall.md >}}) in order for the Scheduler container to be recreated with the new persistent volume.
```

File: daprdocs/content/en/reference/cli/dapr-init.md [hunk 37]
```diff
@@ -55,7 +56,9 @@ dapr init [flags]
 
 **Install**
 
-Install Dapr by pulling container images for Placement, Scheduler, Redis, and Zipkin. By default, these images are pulled from Docker Hub. 
+Install Dapr by pulling container images for Placement, Scheduler, Redis, and Zipkin. By default, these images are pulled from Docker Hub.
+
+> By default, a `dapr_scheduler` local volume will be created for Scheduler to be used as the data base directory. The host file location for this volume is likely located at `/var/lib/docker/volumes/dapr_scheduler/_data` or `~/.local/share/containers/storage/volumes/dapr_scheduler/_data`, depending on your container runtime.
```
Comment (msfussell) [inline daprdocs/content/en/reference/cli/dapr-init.md]:
```suggestion
> By default, a `dapr_scheduler` local volume is created for Scheduler service to be used as the database directory. The host file location for this volume is likely located at `/var/lib/docker/volumes/dapr_scheduler/_data` or `~/.local/share/containers/storage/volumes/dapr_scheduler/_data`, depending on your container runtime.
```

File: daprdocs/content/en/operations/hosting/self-hosted/self-hosted-persisting-scheduler.md [hunk 38]
```diff
@@ -0,0 +1,22 @@
...
+title: "How-to: Persist Scheduler Jobs"
+linkTitle: "How-to: Persist Scheduler Jobs"
+weight: 50000
+description: "Configure Scheduler to persist its database to make it resilient to restarts"
+---
+
+The [Scheduler]({{< ref scheduler.md >}}) service is responsible for writing jobs to its embedded database and scheduling them for execution.
+By default, the Scheduler service database writes this data to the local volume `dapr_scheduler`, meaning that **this data is persisted across restarts**.
+
+The host file location for this local volume will typically be located at either `/var/lib/docker/volumes/dapr_scheduler/_data` or `~/.local/share/containers/storage/volumes/dapr_scheduler/_data`, depending on your container runtime.
```
Comment (msfussell) [inline daprdocs/content/en/operations/hosting/self-hosted/self-hosted-persisting-scheduler.md]:
```suggestion
The host file location for this local volume is typically located at either `/var/lib/docker/volumes/dapr_scheduler/_data` or `~/.local/share/containers/storage/volumes/dapr_scheduler/_data`, depending on your container runtime.
```

**Review by msfussell — COMMENTED** [review]
A few comments, mostly on the default for dapr init command

File: daprdocs/content/en/reference/api/jobs_api.md [hunk 39]
```diff
@@ -46,24 +82,26 @@ Parameter | Description
...
-                "HanSolo": "Running spice"
+	            "@type": "type.googleapis.com/google.protobuf.StringValue",
+	            "value": "Running spice"
             },
-            "dueTime": "30s"
+            "schedule": "@every 1m",
+            "repeats": 5
         }
     }'
 ```
```
Comment (rochabr) [inline daprdocs/content/en/reference/api/jobs_api.md:107]: new request body should be:

```
curl -X POST \
  http://localhost:3500/v1.0-alpha1/jobs/jobforjabba \
  -H "Content-Type: application/json" \
  -d '{
        "data": {
            "@type": "type.googleapis.com/google.protobuf.StringValue",
            "value": "Running spice"
        },
        "schedule": "@every 1m",
        "repeats": 5
    }'
```

**Review by msfussell — APPROVED** [review]
LGTM.

**Review by hhunter-ms — APPROVED** [review]

**Comment by msfussell** [comment]
DO not merge this yet. @JoshVanL will add one more update.

**Comment by JoshVanL** [comment]
@msfussell updated- please take a look!

File: daprdocs/content/en/operations/hosting/self-hosted/self-hosted-persisting-scheduler.md [hunk 40]
```diff
@@ -0,0 +1,27 @@
...
+linkTitle: "How-to: Persist Scheduler Jobs"
+weight: 50000
+description: "Configure Scheduler to persist its database to make it resilient to restarts"
+---
+
+The [Scheduler]({{< ref scheduler.md >}}) service is responsible for writing jobs to its embedded database and scheduling them for execution.
+By default, the Scheduler service database writes this data to the local volume `dapr_scheduler`, meaning that **this data is persisted across restarts**.
+
+The host file location for this local volume is typically located at either `/var/lib/docker/volumes/dapr_scheduler/_data` or `~/.local/share/containers/storage/volumes/dapr_scheduler/_data`, depending on your container runtime.
+Note that if using Docker Desktop, this volume will be located in the Docker Desktop VM's filesystem, which can be accessed using:
```
Comment (msfussell) [inline daprdocs/content/en/operations/hosting/self-hosted/self-hosted-persisting-scheduler.md]:
```suggestion
Note that if you are using Docker Desktop, this volume is located in the Docker Desktop VM's filesystem, which can be accessed using:
```

**Pull request merged**